### PR TITLE
Temporary fix for fetchAllSavedQueriesForSelectedTab() to run saved queries

### PR DESCRIPTION
**Pull request by riysaxen-amzn**
Copied from draft PR -> https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8796

## Testing the changes

```
 yarn test:jest src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.test.tsx
yarn run v1.22.22
$ scripts/use_node scripts/jest src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.test.tsx
jest-haste-map: duplicate manual mock found: index
  The following files share their name; please delete one of them:
    * <rootDir>/src/plugins/vis_type_vega/public/expressions/__mocks__/index.ts
    * <rootDir>/src/core/server/saved_objects/import/__mocks__/index.ts

  console.error
    Warning: Failed prop type: The prop `textToCopy` is marked as required in `EuiCopy`, but its value is `undefined`.
        in EuiCopy (created by SavedQueryCard)
        in SavedQueryCard (created by OpenSavedQueryFlyout)
        in div (created by EuiTabbedContent)
        in div (created by EuiTabbedContent)
        in EuiTabbedContent (created by OpenSavedQueryFlyout)
        in div (created by EuiFlyoutBody)
        in div (created by EuiFlyoutBody)
        in div (created by EuiFlyoutBody)
        in EuiFlyoutBody (created by OpenSavedQueryFlyout)
        in div (created by EuiFlyout)
        in EuiFlyout (created by OpenSavedQueryFlyout)
        in OpenSavedQueryFlyout

      164 |           </EuiFlexItem>
      165 |           <EuiFlexItem grow={false} style={{ marginTop: 0 }}>
    > 166 |             <EuiCopy textToCopy={savedQuery.attributes.query.query as string}>
          |             ^
      167 |               {(copy) => (
      168 |                 <EuiButtonIcon
      169 |                   onClick={copy}

      at console.error (node_modules/@testing-library/react/dist/act-compat.js:55:34)
      at printWarning (node_modules/prop-types/checkPropTypes.js:20:15)
      at checkPropTypes (node_modules/prop-types/checkPropTypes.js:83:11)
      at validatePropTypes (node_modules/react/cjs/react.development.js:1714:7)
      at Object.createElement (node_modules/react/cjs/react.development.js:1813:5)
      at SavedQueryCard (src/plugins/data/public/ui/saved_query_flyouts/saved_query_card.tsx:166:13)
      at renderWithHooks (node_modules/react-dom/cjs/react-dom.development.js:14803:18)
      at mountIndeterminateComponent (node_modules/react-dom/cjs/react-dom.development.js:17482:13)
      at beginWork (node_modules/react-dom/cjs/react-dom.development.js:18596:16)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23179:14)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22157:12)
      at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:22130:22)
      at performSyncWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:21756:9)
      at node_modules/react-dom/cjs/react-dom.development.js:11089:24
      at unstable_runWithPriority (node_modules/scheduler/cjs/scheduler.development.js:653:12)
      at runWithPriority$1 (node_modules/react-dom/cjs/react-dom.development.js:11039:10)
      at flushSyncCallbackQueueImpl (node_modules/react-dom/cjs/react-dom.development.js:11084:7)
      at flushSyncCallbackQueue (node_modules/react-dom/cjs/react-dom.development.js:11072:3)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:21199:9)
      at setHasTemplateQueries (node_modules/react-dom/cjs/react-dom.development.js:15660:5)
      at src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:74:5

  console.warn
    EuiButtonIcon requires aria-label or aria-labelledby to be specified because icon-only
          buttons are screen-reader-inaccessible without them.

      72 |     );
      73 |     setSavedQueries(queriesForSelectedTab);
    > 74 |     setHasTemplateQueries(templateQueriesPresent);
         |     ^
      75 |   }, [savedQueryService, selectedTabId, setSavedQueries]);
      76 |
      77 |   const updatePageIndex = useCallback((index: number) => {

      at EuiButtonIcon (node_modules/@elastic/eui/test-env/eui_components/button/button_icon/button_icon.js:110:13)
      at renderWithHooks (node_modules/react-dom/cjs/react-dom.development.js:14803:18)
      at mountIndeterminateComponent (node_modules/react-dom/cjs/react-dom.development.js:17482:13)
      at beginWork (node_modules/react-dom/cjs/react-dom.development.js:18596:16)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23179:14)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22157:12)
      at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:22130:22)
      at performSyncWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:21756:9)
      at node_modules/react-dom/cjs/react-dom.development.js:11089:24
      at unstable_runWithPriority (node_modules/scheduler/cjs/scheduler.development.js:653:12)
      at runWithPriority$1 (node_modules/react-dom/cjs/react-dom.development.js:11039:10)
      at flushSyncCallbackQueueImpl (node_modules/react-dom/cjs/react-dom.development.js:11084:7)
      at flushSyncCallbackQueue (node_modules/react-dom/cjs/react-dom.development.js:11072:3)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:21199:9)
      at setHasTemplateQueries (node_modules/react-dom/cjs/react-dom.development.js:15660:5)
      at src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:74:5

  console.warn
    EuiButtonIcon requires aria-label or aria-labelledby to be specified because icon-only
          buttons are screen-reader-inaccessible without them.

      72 |     );
      73 |     setSavedQueries(queriesForSelectedTab);
    > 74 |     setHasTemplateQueries(templateQueriesPresent);
         |     ^
      75 |   }, [savedQueryService, selectedTabId, setSavedQueries]);
      76 |
      77 |   const updatePageIndex = useCallback((index: number) => {

      at EuiButtonIcon (node_modules/@elastic/eui/test-env/eui_components/button/button_icon/button_icon.js:110:13)
      at renderWithHooks (node_modules/react-dom/cjs/react-dom.development.js:14803:18)
      at mountIndeterminateComponent (node_modules/react-dom/cjs/react-dom.development.js:17482:13)
      at beginWork (node_modules/react-dom/cjs/react-dom.development.js:18596:16)
      at beginWork$1 (node_modules/react-dom/cjs/react-dom.development.js:23179:14)
      at performUnitOfWork (node_modules/react-dom/cjs/react-dom.development.js:22157:12)
      at workLoopSync (node_modules/react-dom/cjs/react-dom.development.js:22130:22)
      at performSyncWorkOnRoot (node_modules/react-dom/cjs/react-dom.development.js:21756:9)
      at node_modules/react-dom/cjs/react-dom.development.js:11089:24
      at unstable_runWithPriority (node_modules/scheduler/cjs/scheduler.development.js:653:12)
      at runWithPriority$1 (node_modules/react-dom/cjs/react-dom.development.js:11039:10)
      at flushSyncCallbackQueueImpl (node_modules/react-dom/cjs/react-dom.development.js:11084:7)
      at flushSyncCallbackQueue (node_modules/react-dom/cjs/react-dom.development.js:11072:3)
      at scheduleUpdateOnFiber (node_modules/react-dom/cjs/react-dom.development.js:21199:9)
      at setHasTemplateQueries (node_modules/react-dom/cjs/react-dom.development.js:15660:5)
      at src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:74:5

 PASS  src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.test.tsx
  OpenSavedQueryFlyout
    ✓ renders correctly and displays saved queries (225 ms)
    ✓ fetches saved queries on mount (59 ms)

Test Suites: 1 passed, 1 total
Tests:       2 passed, 2 total
Snapshots:   0 total
Time:        5.594 s
Ran all test suites matching /src\/plugins\/data\/public\/ui\/saved_query_flyouts\/open_saved_query_flyout.test.tsx/i.
✨  Done in 10.75s.

```
### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by ashwin-pc** [comment]
@riysaxen-amzn can you add detials about why this change is needed and what the issue is?

**Comment by riysaxen-amzn** [comment]
> @riysaxen-amzn can you add detials about why this change is needed and what the issue is?

 * The existing logic to get all the saved queries is broken in NeoApp so we need this temp fix until a permanent fix is found

**Comment by AMoo-Miki** [comment]
> The existing logic to get all the saved queries is broken in NeoApp so we need this temp fix until a permanent fix is found

Can you please explain what it is that is broken? Looking at the source code, `findSavedQueries` is more restrictive (based on supported languages of the app) than `getAllSavedQueries`. Is that restriction what you need?

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 1]
```diff
@@ -61,7 +61,9 @@ export function OpenSavedQueryFlyout({
   const [searchQuery, setSearchQuery] = useState(EuiSearchBar.Query.MATCH_ALL);
 
   const fetchAllSavedQueriesForSelectedTab = useCallback(async () => {
-    const allQueries = await savedQueryService.getAllSavedQueries();
+    // const allQueries = await savedQueryService.getAllSavedQueries();
+    // TODO: Mitigation until getAllSavedQueries gets fixed
+    const { queries: allQueries } = await savedQueryService.findSavedQueries('', 10000);
```
Comment (virajsanghvi) [inline src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:66]: Is this behind a feature flag or would it be used in either case? Is there a risk with the 10000 limit (I'm assuming)?

Comment (riysaxen-amzn) [reply src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:66]: It is behind flag and we will get it fixed as a fast follow  and we don't see a risk with count as 10k
That's the default upper limit

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 2]
```diff
@@ -61,7 +61,9 @@ export function OpenSavedQueryFlyout({
   const [searchQuery, setSearchQuery] = useState(EuiSearchBar.Query.MATCH_ALL);
 
   const fetchAllSavedQueriesForSelectedTab = useCallback(async () => {
-    const allQueries = await savedQueryService.getAllSavedQueries();
+    // const allQueries = await savedQueryService.getAllSavedQueries();
+    // TODO: Mitigation until getAllSavedQueries gets fixed
```
Comment (joshuali925) [inline src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:65]: what is the issue with `getAllSavedQueries`?

**Comment by AMoo-Miki** [comment]
> The `getAllSavedQueries` returns 0 as total if we pass per page as 0

Then let's not pass `0`. Changing this to `1` should be enough:
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/48bfe2c1790bd213c4fc2bd3f2405ea372249691/src/plugins/data/public/query/saved_query/saved_query_service.ts#L210

**Comment by riysaxen-amzn** [comment]
closing in favor of https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8808/commits

**Pull request closed without merging**